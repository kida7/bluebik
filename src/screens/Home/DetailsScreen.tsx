import React from 'react';
import { StyleSheet, Text, View } from 'react-native';
import Page from '$components/Page';
import { WINDOW_WIDTH } from '$themes/constants';
import {} from 'react-native-gesture-handler';
import { useAppSelector } from '$redux';
import { format } from 'date-fns';

const AVATAR_WIDTH = WINDOW_WIDTH - 16 * 2;

const Field = ({ title, content }: { title: string; content?: string }) => {
  return (
    <View style={styles.fieldContainer}>
      <Text style={styles.fieldTitle}>{title}</Text>
      <Text style={styles.fieldContent}>{content}</Text>
    </View>
  );
};

const DetailsScreen = ({}) => {
  const fullName = useAppSelector(state => state.Onboarding.fullName);
  const idNumber = useAppSelector(state => state.Onboarding.idNumber);
  const email = useAppSelector(state => state.Onboarding.email);
  const phoneNumber = useAppSelector(state => state.Onboarding.phoneNumber);
  const dob = useAppSelector(state => state.Onboarding.dob);
  const purpose = useAppSelector(state => state.Onboarding.purpose);
  return (
    <Page>
      <View style={styles.container}>
        <Text style={styles.header}>Personal Information</Text>
        <Field title="Full Name" content={fullName} />
        <Field title="ID Number" content={idNumber} />
        <Field title="Email" content={email} />
        <Field title="Phone Number" content={phoneNumber} />
        <Field
          title="Date of Birth"
          content={dob ? format(dob, 'dd/MM/yyyy') : ''}
        />
        <Field title="Purpose" content={purpose?.join(',')} />
      </View>
    </Page>
  );
};

export default DetailsScreen;

const styles = StyleSheet.create({
  header: {
    fontSize: 18,
    fontWeight: '600',
    marginBottom: 30,
    textAlign: 'center',
  },
  fieldContent: {
    flex: 2,
    textAlign: 'right',
  },
  fieldTitle: { flex: 1, fontSize: 14, fontWeight: '600' },
  fieldContainer: { flexDirection: 'row', paddingVertical: 12 },
  container: { paddingHorizontal: 16, paddingTop: 30 },
});
