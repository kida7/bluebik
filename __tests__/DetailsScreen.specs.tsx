import React from 'react';
import renderer, { ReactTestRenderer } from 'react-test-renderer';
import { Provider } from 'react-redux';
import { store } from '$redux';
import DetailsScreen from '$screens/Home/DetailsScreen';

const _store = {
  Onboarding: {
    fullName: 'Vinh Le',
    idNumber: '1234',
    email: 'kida7kahp@gmail.com',
    phoneNumber: '0973886038',
    dob: new Date('1990/01/01'),
    purpose: ['loan'],
  },
};
//@ts-ignore
store.getState = () => _store;

describe('Details Screen ', () => {
  let container: ReactTestRenderer;
  beforeAll(() => {
    container = renderer.create(
      <Provider store={store}>
        <DetailsScreen />
      </Provider>,
    );
  });
  it('Details Screeen render correctly', () => {
    expect(container).toMatchSnapshot();
  });
});
