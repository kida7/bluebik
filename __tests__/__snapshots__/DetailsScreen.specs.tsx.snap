// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Details Screen  Details Screeen render correctly 1`] = `
<View
  style={
    Array [
      undefined,
      undefined,
      Object {},
    ]
  }
>
  <View
    style={
      Array [
        undefined,
        Object {},
      ]
    }
  >
    <View
      style={
        Array [
          undefined,
          Array [
            undefined,
            undefined,
          ],
        ]
      }
    >
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <Icon
          color="#012A58"
          name="arrowleft"
          size={24}
        />
      </View>
      <Text
        style={
          Array [
            undefined,
            Object {},
          ]
        }
      />
    </View>
  </View>
  <View>
    <Text>
      Personal Information
    </Text>
    <View>
      <Text>
        Full Name
      </Text>
      <Text>
        Vinh Le
      </Text>
    </View>
    <View>
      <Text>
        ID Number
      </Text>
      <Text>
        1234
      </Text>
    </View>
    <View>
      <Text>
        Email
      </Text>
      <Text>
        kida7kahp@gmail.com
      </Text>
    </View>
    <View>
      <Text>
        Phone Number
      </Text>
      <Text>
        0973886038
      </Text>
    </View>
    <View>
      <Text>
        Date of Birth
      </Text>
      <Text>
        01/01/1990
      </Text>
    </View>
    <View>
      <Text>
        Purpose
      </Text>
      <Text>
        loan
      </Text>
    </View>
  </View>
</View>
`;
